backend_new/test/unit/services/ami-chart.service.spec.ts

```diff
@@ -0,0 +1,503 @@
+        },
+        {
+          percentOfAmi: 1,
+          householdSize: 1,
+          income: 1,
+        },
+        {
+          percentOfAmi: 2,
+          householdSize: 2,
+          income: 2,
+        },
+      ],
+      name: 'ami name 3',
+    };
+
+    expect(await service.create(params)).toEqual({
+      id: 'ami Id 3',
+      name: 'ami name 3',
+      jurisdictions: jurisdictionData,
+      createdAt: date,
+      updatedAt: date,
+      items: [
+        {
+          percentOfAmi: 0,
+          householdSize: 0,
+          income: 0,
+        },
+        {
+          percentOfAmi: 1,
+          householdSize: 1,
+          income: 1,
+        },
+        {
+          percentOfAmi: 2,
+          householdSize: 2,
+          income: 2,
+        },
+      ],
+    });
+
+    expect(prisma.amiChart.create).toHaveBeenCalledWith({
+      include: {
+        jurisdictions: true,
+      },
+      data: {
+        name: 'ami name 3',
+        jurisdictions: {
+          connect: {
+            id: 'example Id',
+          },
+        },
+        items: [
+          {
+            percentOfAmi: 0,
+            householdSize: 0,
+            income: 0,
+          },
+          {
+            percentOfAmi: 1,
+            householdSize: 1,
+            income: 1,
+          },
+          {
+            percentOfAmi: 2,
+            householdSize: 2,
+            income: 2,
+          },
+        ],
+      },
+    });
+  });
+
+  it('testing update() existing record found', async () => {
+    const date = new Date();
+    const jurisdictionData = {
+      id: 'example Id',
+      name: 'example name',
+    };
+    const mockedAmi = mockAmiChart(3, date, jurisdictionData);
+
+    prisma.amiChart.findUnique = jest.fn().mockResolvedValue(mockedAmi);
+    prisma.amiChart.update = jest.fn().mockResolvedValue({
+      ...mockedAmi,
+      name: 'ami name 4',
+      items: [
+        {
+          percentOfAmi: 0,
+          householdSize: 0,
+          income: 0,
+        },
+        {
+          percentOfAmi: 1,
+          householdSize: 1,
+          income: 1,
+        },
+        {
+          percentOfAmi: 2,
+          householdSize: 2,
+          income: 2,
+        },
+        {
+          percentOfAmi: 3,
+          householdSize: 3,
+          income: 3,
+        },
+      ],
+    });
+
+    const params: AmiChartUpdate = {
+      jurisdictions: jurisdictionData,
+      items: [
+        {
+          percentOfAmi: 0,
+          householdSize: 0,
+          income: 0,
+        },
+        {
+          percentOfAmi: 1,
+          householdSize: 1,
+          income: 1,
+        },
+        {
+          percentOfAmi: 2,
+          householdSize: 2,
+          income: 2,
+        },
+        {
+          percentOfAmi: 3,
+          householdSize: 3,
+          income: 3,
+        },
+      ],
+      name: 'ami name 4',
+      id: 'ami Id 3',
+    };
+
+    expect(await service.update(params)).toEqual({
+      id: 'ami Id 3',
```

Comment:
sure, much like before we are updating the name as part of the call, I'm open to making changes to this data though (especially since none of this is stored in the db)

technically the id would be a UUID in the db so any string here would suffice. I can change it to just the number if you prefer though